src/index.js

```diff
@@ -206,11 +206,11 @@ export const createUpdateChart = (instance) => {
       instance.chart = instance.generateChart(props);
     }
 
+    instance.loadData({...data, unload: unloadBeforeLoad});
+
     if (unloadBeforeLoad) {
       instance.unloadData();
     }
```

Comment:
based on the documentation provided, should this `if` statement still exist? wouldn't the call to `instance.loadData` that you've changed handle both the unload and the load of the data?